recipes/si/all/conanfile.py

```diff
@@ -0,0 +1,77 @@
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+from conans.tools import Version
+
+import os
+
+
+class SiConan(ConanFile):
+    name = "si"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/bernedom/SI"
+    description = "A header only c++ library that provides type safety and user defined literals \
+         for handling pyhsical values defined in the International System of Units."
+    topics = ("physical units", "SI-unit-conversion",
+              "cplusplus-library", "cplusplus-17")
+    exports_sources = "CMakeLists.txt"
+    settings = "os", "arch", "compiler", "build_type"
+    no_copy_source = True
+    generators = "cmake"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def _supports_cpp17(self):
+        supported_compilers = [
+            ("gcc", "7"), ("clang", "5"), ("apple-clang", "10"), ("Visual Studio", "15.7")]
+        compiler = self.settings.compiler
+        version = Version(compiler.version)
+        return any(compiler == sc[0] and version >= sc[1] for sc in supported_compilers)
+
+    def package_id(self):
+        self.info.header_only()
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_folder = "SI-{}".format(self.version)
+        os.rename(extracted_folder, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            # Add additional settings with cmake.definitions["SOME_DEFINITION"] = True
+            self._cmake.definitions["SI_BUILD_TESTING"] = False
+            self._cmake.definitions["SI_BUILD_DOC"] = False
+            self._cmake.definitions["SI_INSTALL_LIBRARY"] = True
+            self._cmake.configure(
+                source_folder=self._source_subfolder, build_folder=self._build_subfolder)
+        return self._cmake
+
+    def configure(self):
```

Comment:
Could you move `configure` method before `package_id` method please? It's easier to read and understand recipes when method are listed in same order than conan calling order.